static/src/javascripts-legacy/projects/commercial/modules/article-body-adverts.js

```diff
@@ -46,12 +50,16 @@ define([
             // we must wait for DFP to return, since if the merch
             // component is empty, it might completely change the
             // positions where we insert those MPUs.
-            im.then(waitForMerch).then(addInlineAds);
+            im.then(waitForMerch).then(addInlineAds).then(moduleEnd);
             return im;
         }
 
-        addInlineAds();
+        addInlineAds().then(moduleEnd);
         return Promise.resolve(true);
+
+        function moduleEnd() {
+            performanceLogging.moduleEnd(moduleName);
```

Comment:
Just because here I did not want to curry twice. In any case, I've been thinking about it and it remains the fact that most modules call start and end at the edges, and so we could probably rewrite them as

```
function(moduleName, module) {
  performanceLogging.moduleStart(moduleName);
  module().then(function () {
    performanceLogging.moduleEnd(moduleName);
  });
}
```

... the problem is, what about those modules that don't comply with this pattern (approx. 1/4 of them)? I'm trying to find a nice composable pattern.